src/test/java/org/opensearch/security/dlic/rest/api/AllowlistApiTest.java

```diff
@@ -0,0 +1,281 @@
+                .put(ConfigConstants.SECURITY_COMPLIANCE_HISTORY_INTERNAL_CONFIG_ENABLED, true)
+                .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_TRANSPORT_CATEGORIES, "authenticated,GRANTED_PRIVILEGES")
+                .put(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES, "authenticated,GRANTED_PRIVILEGES")
+                .build();
+        setupWithRestRoles(settings);
+        TestAuditlogImpl.clear();
+
+        // any creds, admin certificate - OK
+        rh.keystore = "restapi/kirk-keystore.jks";
+        testGetAndPut(HttpStatus.SC_OK, true, nonAdminCredsHeader);
+
+        //TESTS THAT 1 READ AND 1 WRITE HAPPENS IN testGetAndPut()
+        final Map<AuditCategory, Long> expectedCategoryCounts = ImmutableMap.of(
+                AuditCategory.COMPLIANCE_INTERNAL_CONFIG_READ, 1L,
+                AuditCategory.COMPLIANCE_INTERNAL_CONFIG_WRITE, 1L);
+        Map<AuditCategory, Long> actualCategoryCounts = TestAuditlogImpl.messages.stream().collect(Collectors.groupingBy(AuditMessage::getCategory, Collectors.counting()));
+
+        assertThat(actualCategoryCounts, equalTo(expectedCategoryCounts));
+    }
+
+    @Test
+    public void testAllowlistInvalidHttpRequestMethod() throws Exception{
+        setup();
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+
+        response = rh.executePutRequest(ENDPOINT + "/allowlist", "{\"enabled\": true, \"requests\": {\"/_cat/nodes\": [\"GE\"],\"/_cat/indices\": [\"PUT\"] }}", adminCredsHeader);
+        assertThat(response.getBody(), response.getStatusCode(), equalTo(HttpStatus.SC_INTERNAL_SERVER_ERROR));
+        assertTrue(response.getBody().contains("\\\"GE\\\": not one of the values accepted for Enum class"));
+    }
+
+    /**
+     * Tests that the PATCH Api works correctly.
+     * Note: boolean variables are not recognized as valid paths in "replace" operation when they are false.
+     * To get around this issue, to update boolean variables (here: 'enabled'), one must use the "add" operation instead.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testPatchApi() throws Exception{
+        setup();
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+
+        //PATCH entire config entry
+        response = rh.executePatchRequest(ENDPOINT + "/allowlist", "[{ \"op\": \"replace\", \"path\": \"/config\", \"value\": {\"enabled\": true, \"requests\": {\"/_cat/nodes\": [\"GET\"],\"/_cat/indices\": [\"PUT\"] }}}]", new Header[0]);
+        Assert.assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+        response = rh.executeGetRequest(ENDPOINT + "/allowlist", adminCredsHeader);
+        assertEquals(response.getBody(),"{\"config\":{\"enabled\":true,\"requests\":{\"/_cat/nodes\":[\"GET\"],\"/_cat/indices\":[\"PUT\"]}}}");
```

Comment:
Please also add checks that the APIs that have been granted/removed are working as expected.  Applies to the other patch actions in this test as well.

E.g. After this point the `/_cat/indices` API should be able to be `PUT` against, but we aren't validating this functionality.

Reply:
The basic functionality of whitelist/allowlist has been validated in `checkGetAndPutAllowlistPermissions(...)`.

Reply:
The [checkGetAndPutAllowlistPermissions](https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/dlic/rest/api/WhitelistApiTest.java#L92-L99) validates both positive and negative cases. If you'd like to add another integrated workflow as you described, let's add to both `AllowlistApiTest` and `WhitelistApiTest` classes in a separate PR?

Reply:
@peternied There're [existing tests](https://github.com/opensearch-project/security/blob/918d9989d36a4c3b3d8bafc69dac6abe4345e7b9/src/test/java/org/opensearch/security/filter/SecurityRestFilterTest.java#L112-L181) validating whitelist functionality.

Reply:
Thats great, can we get a duplicate copy of that test which will cover allow list scenarios?

Note, I think this test should be in `AllowlistApiTest.java` file rather than in the existing SecurityFilter tests.

Reply:
Yes, I can do that.

Reply:
Done